persist/persist.go

```diff
@@ -50,15 +52,25 @@ type safeFile struct {
 
 // Commit renames the file to the intended final filename.
 func (sf *safeFile) Commit() error {
-	return os.Rename(sf.finalName+"_temp", sf.finalName)
+	return os.Rename(sf.Name(), sf.finalName)
 }
 
 // NewSafeFile returns a file that can atomically be written to disk,
 // minimizing the risk of corruption.
 func NewSafeFile(filename string) (*safeFile, error) {
-	file, err := os.Create(filename + "_temp")
+	// Pass TempFile the empty string as a dir to use the default directory for
+	// temporary files (os.TempDir).
+	file, err := ioutil.TempFile("", filepath.Base(filename))
 	if err != nil {
 		return nil, err
 	}
-	return &safeFile{file, filename}, nil
+
+	// Get the absolute path of the filename so that calling os.Chdir inbetween
+	// creating a safeFile and commiting does not change the final file path.
+	absFilename, err := filepath.Abs(filename)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Regardless of whether this PR is accepted or not I think this change could prevent potential bugs if `os.Chdir` is every inadvertantly called between `NewSafeFile` and `safeFile.Commit`.